### DISCUSSION TITLE
Interoperability with Facenet128d/512d

### DISCUSSION THREAD

**Discussion by AlbertoLopSie** [davisking/dlib#2996] [General]
Hi David,

After playing a while with your library, I wonder if the ResNet Model you trained to recognize faces is in any way related to FaceNet128d.  Both map faces to a 128-dimension vector. 

In order to try and measure results from different libs, could your DLib face_recognition functions be used with another model?  I'm tempted to think that the FaceNet128d is a trivial candidate. Is it?

Also, Do you think it would be to hard to modify that functions to work using another model like FaceNet512d?

I'm trying to develop a portable face recognition system able to run on the same models on edge devices (ie Android) and REST services. For edge devices DLib library looks perfect, but for server apps, Python frameworks like [DeepFace](https://github.com/serengil/deepface) look like more easy to use. Deepface allows to switch easily among several recognition models (including DLib) but according to them the more accurate one appears to be FaceNet512d.

Thanks!!

**Comment by davisking** [comment]
It's not related to facenet.  Just does similar things, but there is no relation or compatibility or anything between them.
